src/main/java/io/neonbee/data/DataException.java

```diff
@@ -53,8 +63,21 @@ public DataException(int failureCode) {
      * @param message     the failure message
      */
     public DataException(int failureCode, String message) {
+        this(failureCode, message, Map.of());
+    }
+
+    /**
+     * Create a DataException.
+     *
+     * @param failureCode   the failure code
+     * @param message       the failure message
+     * @param failureDetail the failure detail message to be propagated to the invoker, but should not be exposed to the
+     *                      consumer
+     */
+    public DataException(int failureCode, String message, Map<String, Buffer> failureDetail) {
```

Comment:
Can we use a map of <String, Object> here and make it clear in JavaDoc that the Object must be compatible to the Vert.x event bus codecs which are registered? That way somebody could also just provide a <String, String> as details and doesn't necesarry need to convert it to buffer first and then loose all the context information...